tensorflow/core/util/tensor_bundle/tensor_bundle.cc

```diff
@@ -400,24 +401,29 @@ BundleWriter::BundleWriter(Env* env, StringPiece prefix, const Options& options)
     : env_(env),
       options_(options),
       prefix_(prefix),
-      tmp_metadata_path_(strings::StrCat(MetaFilename(prefix_), ".tempstate",
-                                         random::New64())),
-      tmp_data_path_(strings::StrCat(DataFilename(prefix_, 0, 1), ".tempstate",
-                                     random::New64())),
       out_(nullptr),
       size_(0) {
+  data_path_ = DataFilename(prefix_, 0, 1);
+  metadata_path_ = MetaFilename(prefix_);
+  use_temp_file_ = env_->NeedsTempLocation(prefix_).ok();
+  if (use_temp_file_) {
+    data_path_ = strings::StrCat(data_path_, ".tempstate", random::New64());
+    metadata_path_ =
+      strings::StrCat(metadata_path_, ".tempstate", random::New64());
+  }
+  
```

Comment:
Rather than doing `x = strings:StrCat(x, z)`, it would be preferable to initialize `x` in both branches of the if. This might make it possible to keep the two path variables still as `const`

Reply:
We can't initialize const members if we have if else case (as we can't initialize in the member initializer list). I think the current one is a bit better as the DataFilename(prefix_, 0, 1) stuff does not have to be duplicated if we do it this way. 

Since we can't have const anyway, do you still want to refactor this? I don't have strong feelings, and can switch to that style if you want.

Reply:
There is a minor semantic change as now any method of the class can change `data_path` and `metadata_path`